### DISCUSSION TITLE
Content-based IDs?

### DISCUSSION THREAD

**Discussion by jminor** [OpenAssetIO/OpenAssetIO#468] [General]
Does OpenAssetIO have any opinions or thoughts on content-based ID systems, like for example C4ID? http://www.cccc.io/c4id-whitepaper-u2.pdf

Would something like that fit into the OpenAssetIO URI structure? Are there any docs or proposals about the specifics of the URI structure, or is that meant to be open-ended?

**Comment by foundrytom** [comment]
Hey @jminor - OpenAssetIO sits on layer above - the format of the entity references is entirely determined by the manager itself. There are some notes [here](https://openassetio.github.io/OpenAssetIO/glossary.html#entity_reference). 

We had a chat with the C4 folks, and would love to look at building a `ManagerPlugin` that could resolve a C4 ID to a loadable URL.
